lai_s3/s3.py

```diff
@@ -117,18 +125,82 @@ def _upload_file(
             self,
             bucket: str,
             object: str,
-            filename: Union[L.storage.Path, str]
+            filename: Union[lapp.storage.Path, str]
     ):
         with open(filename, 'rb') as _f:
             self.resource.meta.client.upload_fileobj(
                 Fileobj=_f, Bucket=bucket, Key=object
             )
 
     def run(self, action, *args, **kwargs):
-
         if action == "get_filelist":
             self._get_filelist(*args, **kwargs)
         elif action == "download_file":
             self._download_file(*args, **kwargs)
         elif action == "upload_file":
             self._upload_file(*args, **kwargs)
+        elif action == 'create_dataset':
+            return self._create_dataset(*args, **kwargs)
+
+
+    def _get_s3_items(data, idx, transforms=None, data_type='img', label_map=None):
+        if data_type.lower() == 'img':
+            obj = data[idx]
+            label = obj.key.split('/')[-2]
+            label = label_map[label]
+            img_bytes = obj.get()['Body'].read()
+            img = Image.open(io.BytesIO(img_bytes)).convert('RGB')
+            out = [img, label]
+        elif data_type.lower() == 'tabular':
+            columns = data.columns
+            features = data.loc[idx, columns[0:-1]].values
+            label = df.at[idx, columns[-1]]
+            out = [features, label]
+        else:
+            raise("This component only supports an img or tabular data_type")
+
+        # Apply preprocessing functions on data
+        if transforms is not None:
+            out[0] = transforms(out[0])
+        return out
+
+    def _create_dataset(
```

Comment:
Can you expose the a public function `create_dataset` here?

Reply:
exposed.